### ISSUE TITLE
Using lua module with lsp flag, but completion behavior is weird

### ISSUE THREAD

**Issue by drcxd** [hlissner/doom-emacs#5770]
### What did you expect to happen?

Consider some simple lua code:
```
local a =
{
  [1] = 1,
  [2] = 2,
  [3] = 3,|
}
```
where `|` denotes the position of the cursor. After I typed the `,` before the cursor, I assume there would no completion offered, and if I type enter afterwards, it should insert a new line character and bring me to the next line. 

### What actually happened?

Same situation as above, after I typed the `,` before the cursor, the completion list shows with some snippets options such as "bang" and "comment line". After that, if I keep typing space, the completion list is still there. However, I think after someone typing comma or space, it is not the right occasion to show the snippets completion list.

### Describe your attempts to resolve the issue

I tried to edit the source in `.emacs.d/modules/lang/lua/config.el` and I found there is configurations about company backends.

```
(set-company-backend! 'lua-mode '(company-lua company-yasnippet))
```

I tried to remove the `comapny-yasnippet` but it does not work.

### Steps to reproduce

You have to enable lua module and add lsp flag to it. The LSP server I use is lua-language-server, so you have to install it. Then just create a new lua source code file and write some random lua code, you will find the behavior of completion is so weird that it is so impractical.

### System Information

https://pastebin.com/YjARHxsY

**Comment by ZethJack** [comment]
Having similar issue. I'm getting snippet hint at the end of the line, especially when there's trailing paren or comma.
My guess it's yas plugin giving the suggestions where it shouldn't. For now I've disabled the +lsp flag for lua.

**Comment by Anskrevy** [comment]
Having the same issue as @ZethJack. The value of `+lsp-company-backends` is
```
(:separate company-files company-capf company-dabbrev-code company-dabbrev)
```
Previously I had `company-yasnippet` included but removed it and now I just get dabbrev suggestions.

**Comment by kelvie** [comment]
Yeah, this is super strange behaviour, and it also happens inside table declarations:

When you go to define a new table (at this point I only typed up to the `{`

![image](https://user-images.githubusercontent.com/5893/218827763-eb486ffd-5ae7-4c02-aa7c-b86bdc5c65bf.png)

Then naturally your cursor is between the `{}`, so you want to add a newline to fill in the table, but instead you get this:

![image](https://user-images.githubusercontent.com/5893/218827922-f0ee575f-d2f4-48f7-baac-a581f9d286e7.png)

And this happens all the time -- I have to basically turn off idle completion to not get this. It also happens when adding new entries to an existing table declaration.

I've since added:
```
(setq-hook! lua-mode company-idle-delay 1.0)
```
to my `config.el` to at least make it happen less often. Maybe one option is to not take a completion option if it's flashed on your screen within the last say 250ms? It's not like there's a chance anyone would have read and grokked the completions within that time anyway. Or can we configure `RET` to not take a completion and use `<tab>` instead?
